### PR TITLE
feat: prototype to support multiple prefixes

### PR DESCRIPTION
**Pull request by Bento007**
## Reason for Change

- [Add support to allow multiple obo prefixes per ontology](https://github.com/chanzuckerberg/cellxgene-ontology-guide/issues/223)

## Changes

- add `additional_term_prefix` to ontology_info schema
- check for `additional_term_prefix` in `_extract_ontology_term_metadata`

## Testing steps
- run unit tests
- run schema upgrade

### CODE REVIEW

File: tools/ontology-builder/tests/test_all_ontology_generator.py
Comment (Bento007) [inline tools/ontology-builder/tests/test_all_ontology_generator.py:1]: This test does not currently work.

File: tools/ontology-builder/tests/test_all_ontology_generator.py [hunk 1]
```diff
@@ -224,3 +235,52 @@ def test_deprecate_previous_cellxgene_schema_versions(mock_datetime):
...
+            comment = ["Deprecated term"]
+            IAO_0000233 = ["http://example.org/term_tracker"]
+            IAO_0100001 = ["http://example.org/replaced_by"]
+
+    onto.save(file=str(tmp_path.joinpath("test_ontology.owl")))
+    return onto
+
+
+def test_extract_ontology_term_metadata(sample_ontology):
+    allowed_ontologies = ["FOO"]
```
Comment (Bento007) [inline tools/ontology-builder/tests/test_all_ontology_generator.py:270]: This test doesn't not currently work

**Review by joyceyan — APPROVED** [review]

File: tools/ontology-builder/src/all_ontology_generator.py [hunk 2]
```diff
@@ -266,8 +267,8 @@ def _parse_ontologies(
         version = ontology_info[onto.name]["version"]
         output_file = os.path.join(output_path, get_ontology_file_name(onto.name, version))
         logging.info(f"Processing {output_file}")
-
-        onto_dict = _extract_ontology_term_metadata(onto)
+        allowed_ontologies = [onto.name] + ontology_info[onto.name].get("additional_ontologies", [])
```
Comment (Bento007) [inline tools/ontology-builder/src/all_ontology_generator.py:270]: Should `allowed_ontologies` agree with `additional_ontologies` in the schema?

File: tools/ontology-builder/tests/test_all_ontology_generator.py [hunk 3]
```diff
@@ -224,3 +235,121 @@ def test_deprecate_previous_cellxgene_schema_versions(mock_datetime):
     deprecate_previous_cellxgene_schema_versions(ontology_info, "v1")
 
     assert ontology_info == expected_ontology_info
+
+
+@pytest.fixture
+def sample_ontology(tmp_path):
```
Comment (Bento007) [inline tools/ontology-builder/tests/test_all_ontology_generator.py:241]: @nayib-jose-gloria thank you for fixing.

File: tools/ontology-builder/src/all_ontology_generator.py (hunk 2 above)
Comment (nayib-jose-gloria) [reply tools/ontology-builder/src/all_ontology_generator.py:270]: I think allowed_ontologies == main ontology + additional ontologies makes intuitive sense as-is

File: asset-schemas/ontology_term_id_schema.json [hunk 4]
```diff
@@ -2,6 +2,10 @@
   "$schema": "http://json-schema.org/draft-07/schema#",
   "title": "Ontology Version and Source Schema",
   "definitions": {
+    "AfPO_term_id": {
```
[the remark this answers is not in the document]
Comment (nayib-jose-gloria) [reply asset-schemas/ontology_term_id_schema.json:5]: It doesn't hurt to have an index of all the terms we parse for + explicitly enforce we're not including anything we don't want to. I'm fine to keep it as it is

[the remark this answers is not in the document]
Comment (joyceyan) [reply asset-schemas/ontology_term_id_schema.json:5]: i'm not entirely sure what the alternative would be, but i think i would slightly prefer having an explicit allowlist of prefixes we parse for

**Pull request merged**